core/src/main/java/com/linecorp/armeria/client/ClientRequestContext.java

```diff
@@ -38,6 +41,43 @@
  */
 public interface ClientRequestContext extends RequestContext {
 
+    /**
+     * Returns the context of the {@link Request} that is being handled in the current thread.
+     *
+     * @throws IllegalStateException if the context is unavailable in the current thread or
+     *                               the current context is not a {@link ClientRequestContext}.
+     */
+    static ClientRequestContext current() {
+        final RequestContext ctx = RequestContext.current();
+        checkState(ctx instanceof ClientRequestContext,
+                   "The current context is not a client-side context: %s", ctx);
+        return (ClientRequestContext) ctx;
+    }
+
+    /**
+     * Maps the context of the {@link Request} that is being handled in the current thread.
+     *
+     * @param mapper the {@link Function} that maps the {@link ClientRequestContext}
+     * @param defaultValueSupplier the {@link Supplier} that provides the value when the context is unavailable
+     *                             in the current thread. If {@code null}, the {@code null} will be returned
```

Comment:
Don't we need to explain here when the `ctx` is not a `ClientRequestContext`?
Or I think, how about raising an exception when it does? :-)

Reply:
That's a good idea. Let me raise an exception when the type is not correct.